### PR TITLE
Produce Metaschemas without XXEs (#1665)

### LINKED ISSUE #1665: Generate and Publish OSCAL Metaschema without XML Entities
### User Story

As a NIST or community OSCAL developer, in order to more efficiently develop tooling that can statically or dynamically generate the [Metaschema source files in `./src/metaschema` (at that location at the time of this request)](https://github.com/usnistgov/OSCAL/tree/46700bade18248d1ed965d3d123a01944fa80ac7/src/metaschema) and potentially publish copies without the XML entities to reduce the burden of working with the current XML source version of the Metaschema definitions.

### Goals

- [ ] Minimize or eliminate the burden of developing Metaschema-technology for OSCAL libraries (specifically for oscal-cli-nodejs (usnistgov/oscal-cli-nodejs#21), metaschema-node, and [community libraries hindered by this choice, such as this PR](https://github.com/GoComply/metaschema/pull/10) [discussed here](https://app.gitter.im/#/room/#GoComply_community:gitter.im/$ZKx03meWDzbk_lbWD1ioKWEH2ujqZl8IhUOq8vgypFM) and 
- [ ] Improve compile and run-time security for different NIST and community libraries that must use software that support XML entity resolution

### Dependencies

_No response_

### Acceptance Criteria

- [ ] A spike is performed to determine:
  - [ ] review and select the simplest and fastest solution to implement, XSLT or otherwise (prototype code if necessary, PR merged into repo not mandatory)
  - [ ] write a spec document on how and when in the CI/CD pipeline process this is to be inserted (prototype code if necessary, PR merged into repo not mandatory)
  - [ ] consult the team and determine if the most appropriate or efficient approach is to 1) commit these versions into a directory in the repo adjacent to `./src/metaschema` or 2) prefer another mechanism. Make this decision based on which decision is easier and faster than the other.
- [ ] Draft ADR and get team to review and approve
- [ ] Create follow-on issue to make this work "go to prod" and end up in `main` branch after ADR and spike is complete, before marking this issue as closed or resolved
- [ ] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

### PR DESCRIPTION
**Pull request by nikitawootten-nist**
# Committer Notes

Fixes #1665

Also in this PR:
- Added `list-release-artifacts` utility makefile target for use in OSCAL-Reference;
- Renamed `src/release/README.txt` to a more descriptive name to prevent confusion in the future (see #1891);
- The archive generation target includes the XXE-less metaschema modules.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/usnistgov/OSCAL/blob/main/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

(For reviewers: [The wiki has guidance](https://github.com/usnistgov/OSCAL/wiki/Issue-Review) on code review and overall issue review for completeness.)

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you included examples of how to use your new feature(s)?
- ~~Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.~~

### CODE REVIEW

**Review by wendellpiez — DISMISSED** [review]
Nice concept, nice simple implementation--

File: decisions/0006-source-metaschema-xxes.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Production of transformed source Metaschema modules without external entities (XXEs)
+
+Date: 08/25/2023
+
+## Status
+
+Proposed
```
Comment (aj-stein-nist) [inline decisions/0006-source-metaschema-xxes.md:7]: As I am going to approve this, consider the following suggestion a good sign:

```suggestion
Approved
```

**Review by aj-stein-nist — APPROVED** [review]

File: src/release/release-readme.md [hunk 2]
```diff
@@ -9,8 +9,8 @@ Documentation for the OSCAL models can be found at: https://pages.nist.gov/OSCAL
 
 This release provides 2 types of resources, each located in a different subdirectory:
```
[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply src/release/release-readme.md:10]: So this file is copied to the zip archives as part of the release

[the remark this answers is not in the document]
Comment (aj-stein-nist) [reply src/release/release-readme.md:10]: Sorry given my understanding of that for the first few parts, that's make the last bit hysterical. I thought it was also part of the auto-generated release notes somehow, but I am not sure what is wrong with me. Leave it `.txt` then and I will cross out the last part for posterity?

[the remark this answers is not in the document]
Comment (nikitawootten-nist) [reply src/release/release-readme.md]: Ah I see, I've reverted that change :)

**Review by aj-stein-nist — APPROVED** [review]

**Pull request merged**